src/Simplify_Max.cpp

```diff
@@ -54,13 +54,20 @@ Expr Simplify::visit(const Max *op, ExprInfo *bounds) {
         int lanes = op->type.lanes();
         auto rewrite = IRMatcher::rewriter(IRMatcher::max(a, b), op->type);
 
+        if (rewrite(max(x, x), x) ||
```

Comment:
I don't think max(x, x) can create a new integer overflow that didn't already exist

Reply:
I was worried about changing the order of rewrite rule evaluation - can this first rule be moved down to be evaluated with the rest?